processor/batchprocessor/config.go

```diff
@@ -1,68 +1,69 @@
+	SendBatchMaxSize int `mapstructure:"send_batch_max_size,omitempty"`
 
-	// MetadataKeys is a list of client.Metadata keys that will be
-	// used to form distinct batchers.  If this setting is empty,
-	// a single batcher instance will be used.  When this setting
-	// is not empty, one batcher will be used per distinct
-	// combination of values for the listed metadata keys.
-	//
-	// Empty value and unset metadata are treated as distinct cases.
-	//
-	// Entries are case-insensitive.  Duplicated entries will
-	// trigger a validation error.
-	MetadataKeys []string `mapstructure:"metadata_keys"`
+	// SendBatchSize corresponds to the JSON schema field "send_batch_size".
+	SendBatchSize int `mapstructure:"send_batch_size,omitempty"`
 
-	// MetadataCardinalityLimit indicates the maximum number of
-	// batcher instances that will be created through a distinct
-	// combination of MetadataKeys.
-	MetadataCardinalityLimit uint32 `mapstructure:"metadata_cardinality_limit"`
+	// Timeout corresponds to the JSON schema field "timeout".
+	Timeout time.Duration `mapstructure:"timeout,omitempty"`
 }
 
-var _ component.Config = (*Config)(nil)
-
-// Validate checks if the processor configuration is valid
-func (cfg *Config) Validate() error {
-	if cfg.SendBatchMaxSize > 0 && cfg.SendBatchMaxSize < cfg.SendBatchSize {
-		return errors.New("send_batch_max_size must be greater or equal to send_batch_size")
+// UnmarshalJSON implements json.Unmarshaler.
+func (j *Config) UnmarshalJSON(b []byte) error {
+	var raw map[string]interface{}
+	if err := json.Unmarshal(b, &raw); err != nil {
+		return err
 	}
-	uniq := map[string]bool{}
-	for _, k := range cfg.MetadataKeys {
-		l := strings.ToLower(k)
-		if _, has := uniq[l]; has {
-			return fmt.Errorf("duplicate entry in metadata_keys: %q (case-insensitive)", l)
-		}
-		uniq[l] = true
+	type Plain Config
+	var plain Plain
+	if err := json.Unmarshal(b, &plain); err != nil {
+		return err
+	}
+	if v, ok := raw["metadata_cardinality_limit"]; !ok || v == nil {
+		plain.MetadataCardinalityLimit = 1000.0
+	}
+	if !(plain.MetadataCardinalityLimit >= 0) {
+		return fmt.Errorf("field %s must be >= %d", "metadata_cardinality_limit", 0)
+	}
+	if v, ok := raw["metadata_keys"]; !ok || v == nil {
+		plain.MetadataKeys = []string{}
 	}
-	if cfg.Timeout < 0 {
-		return errors.New("timeout must be greater or equal to 0")
+	if v, ok := raw["send_batch_max_size"]; !ok || v == nil {
+		plain.SendBatchMaxSize = 0.0
+	}
+	if !(plain.SendBatchMaxSize >= 0) {
+		return fmt.Errorf("field %s must be >= %d", "send_batch_max_size", 0)
+	}
+	if v, ok := raw["send_batch_size"]; !ok || v == nil {
+		plain.SendBatchSize = 8192.0
+	}
+	if !(plain.SendBatchSize >= 0) {
+		return fmt.Errorf("field %s must be >= %d", "send_batch_size", 0)
+	}
+	if v, ok := raw["timeout"]; !ok || v == nil {
+		var err error
+		plain.Timeout, err = time.ParseDuration("200ms")
+		if err != nil {
+			return err
+		}
 	}
```

Comment:
Here I cheated a little bit. The original code was:
```go
	if v, ok := raw["timeout"]; !ok || v == nil {
		plain.Timeout = "200ms"
	}
```
I edited it a bit so that it compiles. I will have to update the [upstream PR](https://github.com/omissis/go-jsonschema/pull/257) for `time.Duration` support to generate this kind of code correctly.